### PR TITLE
Expiration and concurrency

### PR DESCRIPTION
**Pull request by VasiliyNovikov**
* Expiration flags for keys
* Concurrency flags for kes and hashes

### CODE REVIEW

File: RedisCore.Tests/RedisClientTests.cs [hunk 1]
```diff
@@ -50,6 +50,37 @@ public async Task RedisClient_Get_Set_Delete_Test(RedisClientConfig config)
...
+        [DynamicData(nameof(Test_Endpoints_Data), typeof(RedisClientTestsBase), DynamicDataSourceType.Method)]
+        public async Task RedisClient_Set_With_Concurrency_Test(RedisClientConfig config)
+        {
+            using (var client = new RedisClient(config))
+            {
+                for (var i = 0; i < 8; ++i)
+                {
+                    var testKey = UniqueString();
+                    var testValue = UniqueString();
+
```
Comment (viktor-ferenczi) [inline RedisCore.Tests/RedisClientTests.cs:63]: Missing corner case?

Trying to delete a set which does not exist:

                    await client.Delete(testKey);
                    Assert.IsNull(await client.GetOrDefault<string>(testKey));

Maybe this does not belong here, since does not test any concurrency flag.

Comment (VasiliyNovikov) [reply RedisCore.Tests/RedisClientTests.cs:63]: We should delete the key to not leave a garbage in the database  but we don't need to check the result since it is tested in different test case

**Review by viktor-ferenczi — APPROVED** [review]
